crates/synthesis/src/two_qubit_decompose.rs

```diff
@@ -2547,7 +2556,14 @@ impl TwoQubitControlledUDecomposer {
                 Python::attach(|py: Python| -> PyResult<(PackedOperation, SmallVec<_>)> {
                     let raw_inverse = gate.gate.call_method0(py, intern!(py, "inverse"))?;
                     let inverse: OperationFromPython = raw_inverse.extract(py)?;
-                    Ok((inverse.operation, inverse.params))
+                    Ok((
+                        inverse.operation,
+                        match inverse.params {
+                            Some(Parameters::Params(params)) => params,
+                            None => smallvec![],
+                            _ => panic!("invalid gate parameters"),
+                        },
```

Comment:
Can we spell out the missing branch a bit more precisely (`Some(Parameters::Blocks(_))`) when the purpose is to produce an "unhandled" panic?  It makes it safer for us to refactor `Parameters` later.